MoviesApp/Model/MoviesListProvider.swift

```diff
@@ -0,0 +1,82 @@
+        Movie(title: "Terrifier 2", rating: 7.1, releaseDate: Date(timeIntervalSinceNow: -1234558), overview: "After being resurrected by a sinister entity, Art the Clown returns to Miles County where he must hunt down and destroy a teenage girl and her younger brother on Halloween night.  As the body count rises, the siblings fight to stay alive while uncovering the true nature of Art's evil intent.", poster: UIImage(), details: Details(duration: 121, genres: [Genre(id: 1, name: "Action")]), popularity: 4608.567),
+        Movie(title: "Black Adam", rating: 6.8, releaseDate: Date(timeIntervalSinceNow: -1234558), isFavourite: true, overview: "Nearly 5,000 years after he was bestowed with the almighty powers of the Egyptian gods—and imprisoned just as quickly—Black Adam is freed from his earthly tomb, ready to unleash his unique form of justice on the modern world.", poster: UIImage(), details: Details(duration: 121, genres: [Genre(id: 1, name: "Action")]), popularity: 3193.354),
+        Movie(title: "Top Gun: Maverick", rating: 8.3, releaseDate: Date(timeIntervalSinceNow: -1234558), isFavourite: true, overview: "After more than thirty years of service as one of the Navy’s top aviators, and dodging the advancement in rank that would ground him, Pete “Maverick” Mitchell finds himself training a detachment of TOP GUN graduates for a specialized mission the likes of which no living pilot has ever seen.", poster: UIImage(), details: Details(duration: 121, genres: [Genre(id: 1, name: "Action")]), popularity: 1219.263)
+    ]
+    
+    func sortBy(moviesToSort: [Movie], criteria: SortCriteria) -> [Movie] {
+        var sortedMovies = moviesToSort
+
+        switch criteria {
+        case .titleAsc: // asc title
+            sortedMovies = moviesToSort.sorted(by: {$0.title < $1.title})
+        case .titleDesc: //desc title
+            sortedMovies = moviesToSort.sorted(by: {$0.title > $1.title})
+        case .ratingDesc: // desc rating
+            sortedMovies = moviesToSort.sorted(by: {$0.rating > $1.rating})
+        case .popularityDesc:// desc popularity
+            sortedMovies = moviesToSort.sorted(by: {$0.popularity > $1.popularity})
+        case .releaseDesc: // desc release date
+            sortedMovies = moviesToSort.sorted(by: {$0.releaseYear > $1.releaseYear})
+        case .none:
+            break
+        }
+        
+        return sortedMovies
+    }
+    
+    func filterBy(moviesToFilter: [Movie], criteria: FilterCriteria) -> [Movie] {
+        var filteredMovies = moviesToFilter
+        
+        switch criteria {
+        case .favourites:
+            filteredMovies = moviesToFilter.filter() { $0.isFavourite == true }
+        case let .rating(value):
+            filteredMovies = moviesToFilter.filter() { $0.rating >= value }
+        case .none:
+            break
+        }
+    
+        return filteredMovies
+    }
+    
+    func sortAndFilter(sortCriteria: SortCriteria, filterCriteria: FilterCriteria) -> [Movie] {
```

Comment:
you should use here to an input of movies (maybe for now have unsortedMovies be a static var and provide it as default value for the movies parameter)..